### ISSUE TITLE
compilation with java 17 fails

### ISSUE THREAD

**Issue by fvalenduc** [squirrel-sql-client/squirrel-sql-code#9]
squirrel-sql can be run with java 17. However, it doesn't seems it compiles with java 17. I get this error:

20:48 francois@pc-francois /usr/src/squirrel-sql/sql12% JAVACMD=/opt/openjdk-bin-17/bin/java ant
Warning: Unable to determine tools.jar location.
  If build fails because sun.* classes could not be found,
  Make sure you are using a JDK, not JRE as your user/system VM.
  and that you have java-config version 2.0.30 or above installed.
Buildfile: /usr/src/squirrel-sql/sql12/build.xml

cleanAllAndInit:
   [delete] Deleting directory /usr/src/squirrel-sql/sql12/output
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/work/bin
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/dist
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/dist/lib
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/dist/plugins
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/izPackInstallJars
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/output/plainZip/work
   [delete] Deleting directory /usr/src/squirrel-sql/sql12/versionChecker/output
    [mkdir] Created dir: /usr/src/squirrel-sql/sql12/versionChecker/output/work/bin

buildExInstaller:

compileVersionCheck:
    [javac] Compiling 1 source file to /usr/src/squirrel-sql/sql12/versionChecker/output/work/bin
    [javac] warning: [options] bootstrap class path not set in conjunction with -source 6
    [javac] error: Source option 6 is no longer supported. Use 7 or later.
    [javac] error: Target option 6 is no longer supported. Use 7 or later.

BUILD FAILED
/usr/src/squirrel-sql/sql12/build.xml:38: The following error occurred while executing this line:
/usr/src/squirrel-sql/sql12/build.xml:117: Compile failed; see the compiler error output for details.

Is this normal ?

**Comment by gerdwagner** [comment]
Yes SQuirreL doesn't compile with java 17 and yes it can currently  be run with all Java versions between 11 and 19.

The usual proceeding is that SQuirreL is built with one LTS version before the latest LTS version.

**Comment by fvalenduc** [comment]
Java 21 is the last LTS version nowadays, so java 17 is the penultimate LTS now. So are there plan to compile squirrel with java 17 now ?

**Comment by gerdwagner** [comment]
Generally I'm willing to upgrade the least supported Java version to 17.
Feel free to provide a patch or merge request with the changes needed to make SQuirreL compile with Java 17.

**Comment by ThomasReinhardt** [comment]
Please see my PR for a trivial fix: #57
